examples/common-sdl.cpp

```diff
@@ -128,11 +128,27 @@ bool audio_async::clear() {
 
         m_audio_pos = 0;
         m_audio_len = 0;
+        m_audio_read_cursor = -1;
+        overran_flag = false;
     }
 
     return true;
 }
 
+bool detect_overrun(size_t old_audio_pos,
+                    size_t new_audio_pos,
+                    size_t read_cursor_pos,
+                    size_t buffer_len) {
+  // We expect new_audio_pos to be the offset into the buffer of the
+  // next sample that will be overridden when new audio comes in,
+  // and read_cursor_pos is the offset of the oldest unread sample.
+  int offset = old_audio_pos;
+  int cursor_point = (read_cursor_pos - offset) % buffer_len;
+  int new_point = (new_audio_pos - offset) % buffer_len;
```

Comment:
I guess a simple way to avoid this is to shift all calculations by buffer_len, ie

```
int cursor_point = (buffer_len + read_cursor_pos - offset) % buffer_len;
int new_point = (buffer_len + new_audio_pos - offset) % buffer_len;
```